lib/src/client.dart

```diff
@@ -24,7 +25,7 @@ class NextCloudClient {
       httpClient,
     );
 
-    _webDavClient = WebDavClient(baseUrl, network);
+    _webDavClient = WebDavClient(baseUrl, network, username);
```

Comment:
A possible solution would be to query the `User` endpoint before creating the `WebDavClient`.

Reply:
Or while creating the WebDav client.